langstream-core/src/main/java/ai/langstream/impl/parser/ModelBuilder.java

```diff
@@ -126,18 +149,62 @@ public static ApplicationWithPackageInfo buildApplicationInstance(
                                         e);
                             }
                         }
+                    } else {
+                        if (Files.isDirectory(path)) {
+                            String filename = path.getFileName().toString();
+                            if (filename.equals("python")) {
+                                recursiveAppendFiles(sortedPyBinaries, path, path);
+                            } else {
+                                if (filename.equals("java")) {
+                                    final Path lib = path.resolve("lib");
+                                    if (Files.exists(lib) && Files.isDirectory(lib)) {
+                                        recursiveAppendFiles(sortedJavaBinaries, lib, lib);
+                                    }
+                                }
+                            }
+                        }
                     }
                 }
             }
         }
+        final ApplicationWithPackageInfo applicationWithPackageInfo =
+                buildApplicationInstance(applicationContents, instanceContent, secretsContent);
+        if (!sortedJavaBinaries.isEmpty()) {
+            final byte[] data = binariesDigestMapper.writeValueAsBytes(sortedJavaBinaries);
```

Comment:
this is not very efficient

it is better to traverse the directory and compute the checksum while reading the files,
you don't need to store the files in memory.
therefore, the algorithm will be more efficient.

you can open a MessageDigest and then pass the contents of the files all in one